controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -393,6 +394,14 @@ private SegmentRange convert(final String scope,
                 .thenApplyAsync(res -> TxnState.newBuilder().setState(TxnState.State.valueOf(res.name())).build(), executor);
     }
 
+    public CompletableFuture<StreamStatus> getStreamState(final String scope,
+                                                         final String stream) {
+        Exceptions.checkNotNullOrEmpty(scope, "scope");
+        Exceptions.checkNotNullOrEmpty(stream, "stream");
+        return streamStore.getState(scope, stream, true, null, executor)
```

Comment:
this can give you states like SCALING etc. You need to perform a translation of such transient workflow induced states into ACTIVE

Reply:
thanks @shiveshr,  this has been addressed in latest commit